### PR TITLE
Implement help command

### LINKED ISSUE #59: Edit "help" Command to display commands available
To match the help command we have stated in the UG

### LINKED ISSUE #77: Add tests for help method

### PR DESCRIPTION
**Pull request by JoeyChenSmart**
Closes #76 

Automatically generate help using the already existing information in the Command's parameters.

## Changes to commands
Added a new (implicitly) required constant String for each command, `SHORT_DESCRIPTION` which will be used in generating `help`.

We will follow the same convention as commits for these descriptions.

Will use this value when adding new commands to the `PrimitiveCommandParser` (where you have to add it is obvious when viewing the code).

## Future todos?
Will have to add docs to the DG for how this works.

The UI `help` button still works and shows the old help popup. May have to remove this but we can do this in a separate issue.

### CODE REVIEW

**Review by dcchan98 — APPROVED** [review]
LGTM

File: src/main/java/jimmy/mcgymmy/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
  */
 public class AddCommand extends Command {
     public static final String COMMAND_WORD = "add";
-
+    public static final String SHORT_DESCRIPTION = "adds a new food item to McGymmy";
```
Comment (aidoxe-123) [inline src/main/java/jimmy/mcgymmy/logic/commands/AddCommand.java]: I think the first letter of the string should be capitalized to be the same format as other files

**Review by Jh123x — APPROVED** [review]
LGTM!
Works on My Windows Machine too

**Comment by JoeyChenSmart** [comment]
Will make this a not draft after I add some tests or something maybe

The code is mostly formatting text to be printed so I dont think unit tests will benefit it much

**Comment by Jh123x** [comment]
> Will make this a not draft after I add some tests or something maybe
> 
> The code is mostly formatting text to be printed so I don't think unit tests will benefit it much

Might need to change the logger message in ui/HelpWindow.java and remove the function that shows the help screen

File: src/main/java/jimmy/mcgymmy/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -24,6 +24,7 @@
  */
 public class EditCommand extends Command {
     public static final String COMMAND_WORD = "edit";
+    public static final String SHORT_DESCRIPTION = "Edit the selected food item.";
```
Comment (aidoxe-123) [inline src/main/java/jimmy/mcgymmy/logic/commands/EditCommand.java:27]: Inconsistent verb tense: In some files, the string is present simple tense whereas, in other files, it is infinitive.

**Review by aidoxe-123 — COMMENTED** [review]
I think it only needs some minor changes. Other than that, everything LGTM

**Review by Jh123x — APPROVED** [review]
LGTM

**Pull request merged**